log/experimental_level/level.go

```diff
@@ -147,10 +63,87 @@ func (l *logger) Log(keyvals ...interface{}) error {
 	return l.next.Log(keyvals...)
 }
 
-func makeSet(a []string) map[string]struct{} {
-	m := make(map[string]struct{}, len(a))
-	for _, s := range a {
-		m[s] = struct{}{}
-	}
-	return m
+// Option sets a parameter for the leveled logger.
+type Option func(*logger)
+
+// AllowAll is an alias for AllowDebugAndAbove.
+func AllowAll() Option {
+	return AllowDebugAndAbove()
+}
+
+// AllowDebugAndAbove allows all of the four default log levels.
+func AllowDebugAndAbove() Option {
+	return allowed(levelDebug | levelInfo | levelWarn | levelError)
+}
```

Comment:
I would love to see the exported function names get less verbose. In particular I think the current _AndAbove_ terminology is not always intuitive because I often think of level error as _more critical_ than level info and therefore above it in some way. I am not sure that everyone has the same intuition about which levels are the "high" levels and which are the "low" levels. I've seen both philosophies implemented in various logging libraries.

I do think it is generally accepted that level info is a superset of level warn which is a superset of level error from an output perspective. I think that intuition is what our exported API should lean on.

I am not too concerned about the implementation details. I am fine either way. If we can measure a consistent performance difference I would be inclined to go with the faster implementation along with an appropriate level of justification in internal comments.